src/cpp/hu.bme.mit.inf.modes3.ESP32-sensor/railsensor_3/ESP32-sensor/lib/InfraSensor/InfraSensor.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef SENSOR_PRELLTIME
+    #define SENSOR_PRELLTIME 0.01
+#endif
+
+#ifndef INFRASENSOR_HPP
+#define INFRASENSOR_HPP
+
+double SecTime(){
+    double ret=micros()/1000000.0;
+    return ret;
+}
+
+enum InfSensorState{
+  Wait,
+  Detect,
+  Prell,
+  Error
+};
+  
+class InfSensor{
+private:
+  const int prelltime;
+  const int pin;
+  int value;
+  InfSensorState state;
+  int sindex;
+public:
+  double DATAstack[20];
```

Comment:
Also, should this field be `public`? Public fields always imply a gut feeling, because in this way anyone with malicious intensions, may modify this field, without the object (who encapsulates it) knowing it.